students/piechowski.michal/lesson_04_unit_testing/collatz_tests.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import unittest
+parent_dir_name = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(parent_dir_name + "/lesson_03_functions")
+
+from collatz_sequence import collatz
```

Comment:
Not possible to move this up because then import wouldnt' succeed. If you want I can do dummy copy of file from lesson3 and put it to lesson4 but I would suggest to ingore stickler message

Reply:
Nie zapomnij dodać `if __name__ == "__main__":`, żeby nie było problemu podczas wykonywania testów i imporotwaniu modułu

Reply:
Hej,

"__main__" mam pododawane, takze tu nie ma problemu, ale zastanawiam się jak miałby działać ten import, o którym wspomniałeś. Zmianę nazwy folderu rozumiem i nawet dla testu tak zrobiłem, ale ten import nie zadziała, chyba że coś pomijam, albo czegoś nie rozumiem.

Generalnie plik jest tutaj:
students/piechowski_michal/lesson_04_unit_testing/collatz_tests.py
i w tym pliku dałem taki import jak napisałeś, ale z perspektywy tego pliku nie istnieje coś takiego jak "students" (ImportError: No module named 'students'). Owszem gdybym do lekcji 4 skopiował tą strukturę katalogów z lekcją 3 to by to pewnie zadziałało, ale moje rozwiązanie ma tę zaletę, że pozwala unikąć kopiowania plików, więc nawet jeśli to nie jest zalecane to i tak chyba wygląda lepiej niż "kopiuj wklej" plików z poprzedniej lekcji do środka obecnej lekcji.
W kazdym razie jezeli cos pomijam i da sie w pythonie zrobic "relative path" import to daj znać bo chętnie bym tak zrobił bez potrzeby używania tego modułu os.

Reply:
jeżeli faktycznie masz z tym problem to zostaw tak jak jest. U mnie lokalnie działa bez problemu, PyCharm sam sugeruje import z tamtej lokalizacji, na ta chwilę nie wiem gdzie leży u ciebie problem. ja jako projekt zaimportowałem całe repo, nie tylko swój katalog roboczy.

Reply:
Jeszcze możesz zobaczyć czy jak dodasz do folderów z kodem Pythona plik __init__.py, czy ci zadziała.

Reply:
Ok, to moze byc kwestia odpowiedniego zaimportowania do PyCharma, sprawdze to. Dzięki :)

Reply:
W PyCharmie dziala, takze poprawie to i inne uwagi